spec/openapi.yaml

```diff
@@ -1,50 +1,245 @@
 openapi: 3.0.1
+
+servers:
+- url: https://resources.operationcode.org/api/v1
+  description: Default Server
 info:
   title: Operation Code Resources API
-  description: A collection of programming and cyber security learning resources for
-    the Operation Code community.
+  version: '1.0'
+  x-logo:
+    url: 'https://operation-code-assets.s3.us-east-2.amazonaws.com/branding/logos/small-blue-logo.png'
+    altText: 'Operation Code Logo'
+    backgroundColor: '#252e3e'
+    href: 'https://operationcode.org/contact'
+  description: |
+    # Introduction
+    Operation Code Resources API is a collection of programming and cyber security learning resources curated by and for the Operation Code community.
+
+    # Authentication
+    Create (POST) and update (PUT) routes are protected and require an API key. API key requests to the `/apikey` endpoint require valid Operation Code credentials in the request body. 
+    
+    If you are not already a member of our community, join Operation Code [HERE](https://operationcode.org/join).
+
+    When you create a request to an authenticated route, you *MUST* include the header `x-apikey` along with your API key.
+
+
   contact:
     name: Operation Code
     url: https://operationcode.org/contact
-  version: "0.1"
-servers:
-- url: https://resources.operationcode.org/api/v1
+
 paths:
+  /apikey:
+    post:
+      summary: API Key Request
+      description: Request API key by providing valid Operation Code credentials
+      tags: 
+        - Request Key
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/ApiKeyRequest'
+        required: true
+      responses:
+        200:
+          description: API key response
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ApiKeyResponse'
+              example: |
+                {
+                  'apiVersion': '1.0',
+                  'data': {
+                    'apikey': 'yourapikey',
+                    'email': 'your@email.com'
+                  }
+                  'status': 'ok',
+                  'status_code': 200
+                }
+        401:
+          $ref: '#/components/responses/Unauthorized'
+        400:
+          $ref: '#/components/responses/BadRequest'
```

Comment:
Let's add a `description` here that states that this 400 Bad Request happens when you don't include a body with your request. That will help folks troubleshoot what the problem is when they encounter this

Reply:
I'm so sorry to change this up on you last minute, but I came across this in the code for the validate resource function
https://github.com/OperationCode/resources_api/blob/aa3e33d4ca07c4ff929922c990297f0ad3b160a0/app/utils.py#L157-L160

Basically, it returns a 422 with a message stating it requires a JSON body to use this endpoint and I think that's way better than returning a 400 with a cryptic message about "expected start character" or whatever... So let's document this as a 422 and a 401, and I'm going to open an issue in the Resources API to catch that error and return a proper message should someone forget to include a body with their request

Reply:
https://github.com/OperationCode/resources_api/issues/199